Sources/Regex/Parse/Token.swift

```diff
@@ -41,16 +44,23 @@ extension Token {
     case rsquare = "]"
     case minus = "-"
     case caret = "^"
-    case dollar = "$"
   }
 
-
   enum SetOperator: String, Hashable {
     case doubleAmpersand = "&&"
     case doubleDash = "--"
     case doubleTilda = "~~"
   }
 
+  enum SpecialCharacterEscape: String, Hashable {
+    case tab = "\\t"
+    case carriageReturn = "\\r"
+    case backspace = "\\b"
+    case formFeed = "\\f"
+    case bell = "\\a"
+    case escape = "\\e"
+  }
```

Comment:
Why not `Character`, which would synthesize the switch for you?

Reply:
It felt weird to just have the raw values be e.g `t` without the `\`, and there's not many of them so it didn't really feel like much of a chore to spell out. I'm happy to change it tho.